frappe/core/doctype/user_type/user_type.py

```diff
@@ -234,7 +234,7 @@ def get_user_linked_doctypes(doctype, txt, searchfield, start, page_len, filters
 		"DocType",
 		fields=["`tabDocType`.`name`"],
 		filters=filters,
-		order_by="`tabDocType`.`idx` desc",
```

Comment:
We can probably just implement this in qb_query.py parse the expression and convert it into `DocType(dt_name).field`